src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/validations/OrderValidator.java

```diff
@@ -0,0 +1,42 @@
+package com.wojciechbarwinski.demo.epic_board_games_shop.validations;
+
+import com.wojciechbarwinski.demo.epic_board_games_shop.dtos.OrderLineDTO;
+import com.wojciechbarwinski.demo.epic_board_games_shop.entities.Product;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class OrderValidator {
+
+    public void validateOrderLineDTOS(List<OrderLineDTO> orderLineDTOs, Map<Long, Product> productsFromDB) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        for (OrderLineDTO orderLineDTO : orderLineDTOs) {
+            validateOrderLineDTO(productsFromDB, orderLineDTO, errors);
+        }
+
+        if (!errors.isEmpty()) {
+            throw new ValidationException(errors);
+        }
+    }
+
+    private void validateOrderLineDTO(Map<Long, Product> productsFromDB, OrderLineDTO orderLineDTO, List<ValidationError> errors) {
+        Long productId = orderLineDTO.productId();
+        Product productById = productsFromDB.get(productId);
+
+        if (productById == null) {
+            errors.add(new ValidationError(
+                    "product Id",
+                    "There is no product with this id",
+                    orderLineDTO.productId()));
+        } else if (productById.getQuantity() <= orderLineDTO.quantity()) {
+            errors.add(new ValidationError(
+                    "product quantity",
+                    String.format("There is insufficient quantity for product with id %d", productId),
+                    String.format("Ordered quantity = %d, current quantity = %d", orderLineDTO.quantity(), productById.getQuantity())));
```

Comment:
Tutaj powinna teoretycznei być konkretna wartość, jaka jest niewłaściwa (order quantity)

Reply:
W terorii tak, ale zastanowiłem się co to daje za informacje zwrotną... żadną. Nadal nie wiem ile realnie jest w magazynie i ile mogę zamówić. Wiec wydaje mi sie że "biznesowo" lepiej wysłać też ile obecnie realnie jest i nie kazać strzelać na oślep.

Reply:
To wyobraź sobie taki scenariusz - Twoje API jest konsumowane przez jakiś front i front chce wywalić ten komunikat userowi nad polem formularza z produktem, którego to się tyczy. Wszystko fajnie, póki front jest po angielsku, ale jak miałby być w innym języku to się robią schody :D Idealnie byłoby tutaj zwrócić  dwie wartości upakowane w obiekt - actual i onStock... ale... actual mamy w requeście, który front wysłał :P Więc czysto technicznie wystarczyłby nam "onStock" 🤔